### DISCUSSION TITLE
对于印章识别中部分配置文件不对应

### DISCUSSION THREAD

**Discussion by sadays22** [PaddlePaddle/PaddleOCR#13567] [Q&A]
### Search before asking

- [X] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.

- [X] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.

- [X] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### Bug

![error20](https://github.com/user-attachments/assets/408b284c-a4cd-4203-bda8-b9c292c3d336)

### Environment

windows11 paddlepaddle-gpu          3.0.0b1       cuda12.5

### Minimal Reproducible Example

    with open(save_res_path, "wb") as fout:
        for file in get_image_file_list(config["Global"]["infer_img"]):
            logger.info("infer_img: {}".format(file))
            with open(file, "rb") as f:
                img = f.read()
                data = {"image": img}
            batch = transform(data, ops)
            images = np.expand_dims(batch[0], axis=0).astype(np.float32)
            shape_list = np.expand_dims(batch[1], axis=0)
            # images = images.astype(np.float32)
            # images = paddle.to_tensor(images)
            images = paddle.to_tensor(images)
            preds = model(images)
改变了tools/infer_e2e中的部分文件用读入float32的数据格式，但是之后PGnet中的参数似乎有点问题

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by GreatV** [comment]
字典加载得不对，pgnet 是在英文上训练的，你加载了中文字典。

**Comment by Akiramex** [comment]
你有试过两段式的识别和推理吗，我用下载下来的DB++推理模型测试出来识别不了印章

**Comment by sadays22** [comment]
> 字典加载得不对，pgnet 是在英文上训练的，你加载了中文字典。

这个是官方给的配置文件，我应该改变这个字典吗
我使用的模型和配置都是官方训练好的pgnet配置文件和模型，只有推理没有训练 使用的命令行是python tools/infer_e2e.py -c configs/pgnet_r50_vd_seal/config.yml -o Global.pretrained_model=./pgnet_r50_vd_seal/best_accuracy.pdparams Global.infer_img="C:\Users\3803
8\Desktop\shiyan\sucaishibie\01.png"
![1](https://github.com/user-attachments/assets/af4b329d-2be3-4594-abd0-810a80e223af)

**Comment by sadays22** [comment]
> 你有试过两段式的识别和推理吗，我用下载下来的DB++推理模型测试出来识别不了印章

没 我只试了ppyolo来检测文章位置和pgnet做端到端的文字识别

**Comment by sadays22** [comment]
https://github.com/PaddlePaddle/PaddleOCR/issues/11986
和这个问题应该是一样的，但是作者没给出解答。。希望有大佬能解答一下

**Comment by Akiramex** [comment]
我和你差不多，ppyolo检测印章没问题，很准确。但是印章识别不行，pgnet直接运行报错，两段式的检测不出来，把DB++转为推理模型用predict_det也一样

**Comment by GreatV** [comment]
https://paddlepaddle.github.io/PaddleOCR/algorithm/end_to_end/algorithm_e2e_pgnet.html pgnet是在英文数据集上训练的，检测不了中文。

**Comment by GreatV** [comment]
你得自己训练一个中文的pgnet模型。

**Comment by Akiramex** [comment]
https://aistudio.baidu.com/projectdetail/6768694?channelType=0&channel=0
模型是这里获取的，按理来说是按照中文训练的模型呀

**Comment by sadays22** [comment]
> 你得自己训练一个中文的pgnet模型。

但是他官方给了一个pgnet的模型参数和配置，所以说官方给的这个不行吗，上面那个链接上写的是可以直接用

**Comment by sadays22** [comment]
> https://aistudio.baidu.com/projectdetail/6768694?channelType=0&channel=0 模型是这里获取的，按理来说是按照中文训练的模型呀

是的，你问题和我应该一样，看来官方这个写的可能有点问题，就是用不了么

**Reply by SWHL** [reply]
用不了的，pgnet给的是检测和识别英文的文字。从pgnet issue来看，训练中文的pgnet难度很大。

**Reply by sadays22** [reply]
> 用不了的，pgnet给的是检测和识别英文的文字。从pgnet issue来看，训练中文的pgnet难度很大。

所以最好是用两阶段的么

**Reply by SWHL** [reply]
如果想做印章识别，建议用TrOCR(可以参考[这个](https://github.com/Gmgge/TrOCR-Seal-Recognition))来做，这种比较适合做二维的文字识别。
不建议两阶段做印章。涉及到环形文字检测，以及环形文字拉直，拉直后的文字识别，这三步都很麻烦。

**Reply by sadays22** [reply]
> 如果想做印章识别，建议用TrOCR(可以参考[这个](https://github.com/Gmgge/TrOCR-Seal-Recognition))来做，这种比较适合做二维的文字识别。 不建议两阶段做印章。涉及到环形文字检测，以及环形文字拉直，拉直后的文字识别，这三步都很麻烦。

这边有这个一对一的标注的数据集吗，，好像这个文档比较简略，怎么做相对应的数据集也没说，大佬有空帮一下，第一次做自己训练的模型

**Reply by sadays22** [reply]
> 如果想做印章识别，建议用TrOCR(可以参考[这个](https://github.com/Gmgge/TrOCR-Seal-Recognition))来做，这种比较适合做二维的文字识别。 不建议两阶段做印章。涉及到环形文字检测，以及环形文字拉直，拉直后的文字识别，这三步都很麻烦。

那边好像只有识别位置的一对一标注，我已经有了，就差文字识别的一对一格式的数据集了

**Reply by SWHL** [reply]
这种去网上搜一下吧，我这里也没有

**Comment by GreatV** [comment]
你得自己训练一下
